tests/src/pathFollowingAdaptiveStepSizingTest.cpp

```diff
@@ -0,0 +1,182 @@
+// SPDX-FileCopyrightText: 2022 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+//
+#include <config.h>
+
+#include "testHelpers.hh"
+
+#include <dune/common/test/testsuite.hh>
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#include <ikarus/assembler/simpleAssemblers.hh>
+#include <ikarus/controlRoutines/pathFollowingTechnique.hh>
+#include <ikarus/linearAlgebra/nonLinearOperator.hh>
+#include <ikarus/utils/drawing/griddrawer.hh>
+#include <ikarus/utils/functionSanityChecks.hh>
+#include <ikarus/utils/init.hh>
+#include <ikarus/utils/observer/nonLinearSolverLogger.hh>
+
+using namespace Ikarus::Concepts;
+using Dune::TestSuite;
+
+auto residual(const Eigen::VectorXd& D, double lambda) {
+  Eigen::VectorXd vec;
+  vec.resize(2);
+  auto& w = D[1];
+  auto& u = D[0];
+  vec << 16.0 * u - 3 * w - w * w * w - (2.0 / 3.0) * lambda, -3 * u + 4 * w - 3 * u * w * w - (9.0 / 4.0) * lambda;
+  return vec;
+}
+auto stiffnessMatrix(const Eigen::VectorXd& D, [[maybe_unused]] double lambda) {
+  Eigen::MatrixXd mat;
+  mat.setZero(2, 2);
+  auto& w = D[1];
+  auto& u = D[0];
+  mat << 16.0, -3.0 * w * w - 3, -3 * w * w - 3, -6 * u * w + 4;
+  return mat;
+}
+
+auto simple2DOperatorArcLengthTest() {
+  double lambda = 0;
+  Eigen::VectorXd D;
+  D.setZero(2);
+
+  auto fvLambda  = [&](auto&& D_, auto&& lambda_) { return residual(D_, lambda_); };
+  auto dfvLambda = [&](auto&& D_, auto&& lambda_) { return stiffnessMatrix(D_, lambda_); };
+
+  auto nonLinOp
+      = Ikarus::NonLinearOperator(Ikarus::linearAlgebraFunctions(fvLambda, dfvLambda), Ikarus::parameter(D, lambda));
+
+  auto linSolver = Ikarus::ILinearSolver<double>(Ikarus::SolverTypeTag::d_LDLT);
+
+  double stepSize = 0.1;
+  int load_steps  = 50;
+  bool isAdaptiveStep = true;
+  double refIte = 3;
+
+  auto pft = Ikarus::StandardArcLength{};  // Path following type
+

```

Comment:
For me it is not clear how the correct stepsizing is tested. Here, it is sucessful if the PathFolloiwng works but it can also work wrongly. We can discuss how this should be really tested

Reply:
This test is removed. A separate test will be written with Kirchhoff-Love shell elements that tests the step sizing vigorously.